dbt-snowflake/src/dbt/include/snowflake/macros/adapters.sql

```diff
@@ -150,7 +150,7 @@
     {# -- Gated for performance reason. If you don't want Iceberg, you shouldn't pay the
        -- latency penalty. #}
     {% if adapter.behavior.enable_iceberg_materializations.no_warn %}
-      select all_objects.*, is_iceberg
+      select all_objects.*, is_iceberg as is_iceberg_dbt
```

Comment:
I think we need a comment here. I needed to re-read this whole thing a few times before I realized that the issue was that `all_objects` contains `is_iceberg` after a particular bundle is enabled. This goes back to us not conditioning on platform version (e.g. bundle), though that's certainly out of scope here.